### ISSUE TITLE
Allow option to use fused linear cross entropy during validation

### ISSUE THREAD

**Issue by connermanuel** [linkedin/Liger-Kernel#703]
### 🚀 The feature, motivation and pitch

Currently, the `forward` methods of various models are hardcoded so that the fused linear cross entropy loss is only used if (a) `model.training` is True and (b) labels are provided. This means that in any other scenario, fused linear cross entropy is not used - instead, the logits are materialized and, if labels are provided, regular cross entropy is used. 

This can lead to memory spikes in validation, which can be annoying especially if the user doesn't require logits, e.g.:

![Image](https://github.com/user-attachments/assets/6acc6f0f-4b3d-46ec-b0fe-8d2f2d82f337)

My proposal is to add a kwarg to the model forward methods that allows users to explicitly choose whether they want to use the fused kernel or if they want to materialize logits and use regular CE loss. This kwarg should be optional: if not provided, default to the original behavior.

Basically, changing this:
```
if self.training and labels is not None:
    loss = LigerForCausalLMLoss(...)
else:
    logits = forward(...)
    loss = CrossEntropy(logits, ...)
```
into this:
```
# kwarg: skip_logits: Optional[bool]

if skip_logits is None:
    skip_logits = self.training and labels is not None

if skip_logits:
    loss = LigerForCausalLMLoss(...)
else:
    logits = forward(...)
    loss = CrossEntropy(logits, ...)
```

If this sounds good I'll make the PR for it.

**Comment by shivam15s** [comment]
Sounds good to me. Feel free to create a PR

**Comment by shimizust** [comment]
Thanks for adding this!
